cmds/info.c

```diff
@@ -209,19 +210,21 @@ static int fill_cmdline(void *arg)
 	struct fill_handler_arg *fha = arg;
 	char *buf = fha->buf;
 	FILE *fp;
+	int err;
 	size_t ret;
-	int i;
+	size_t i;
 	char *p;
 
 	fp = fopen("/proc/self/cmdline", "r");
 	if (fp == NULL)
 		return -1;
 
 	ret = fread(buf, 1, sizeof(fha->buf), fp);
+	err = ferror(fp);
 	fclose(fp);
 
-	if (ret != sizeof(fha->buf))
-		return ret;
+	if (err || ret > INT_MAX)
```

Comment:
I think `if (!ret && err)` is better.

Reply:
ret > INT_MAX is meant to detect fread() return value bigger than int size
Maybe I could change ret to int then check err || ret < 0

Reply:
Hmm.. I think ret cannot be bigger than `sizeof(fha->buf)`.

Reply:
Oh you are right. I will fix as you said

Reply:
Return value of 0 itself is not an error.  I think we should check err with `&&`.

Reply:
Thanks I'll FIx